pkg/artifactory/resource/security/resource_artifactory_trusted_key.go

```diff
@@ -0,0 +1,154 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const trustedKeysAPIEndPoint = "artifactory/api/security/keys/trusted"
+
+type trustedKeyPayLoad struct {
+	KeyID       string `hcl:"key_id" json:"kid"`
+	Alias       string `hcl:"alias" json:"alias"`
+	Fingerprint string `hcl:"fingerprint" json:"fingerprint"`
+	PublicKey   string `hcl:"public_key" json:"key"`
+	IssuedOn    string `hcl:"issued_on" json:"issued_on"`
+	IssuedBy    string `hcl:"issued_by" json:"issued_by"`
+	ValidUntil  string `hcl:"valid_until" json:"valid_until"`
+}
+
+func ResourceArtifactoryTrustedKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: addTrustedKey,
+		DeleteContext: deleteTrustedKey,
+		ReadContext:   readTrustedKey,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "Manage the GPG trusted keys used to verify distributed release bundles",
+
+		Schema: map[string]*schema.Schema{
+			"key_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Generated by artifactory on upload",
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Will be used as a identifier when uploading/retrieving the public key via REST API",
+				ForceNew:    true,
```

Comment:
Please add `ValidateFunc: validation.StringIsNotEmpty` to ensure `alias` is not empty.